app/modules/enduser_setup.c

```diff
@@ -1674,6 +1957,13 @@ static int enduser_setup_init(lua_State *L)
     ENDUSER_SETUP_DEBUG("enduser_setup_init: Debug callback has been set");    
   }
 
+  if (!lua_isnoneornil(L, 4))
+  {
+    lua_pushvalue (L, 4);
+    state->lua_validation_cb_ref = luaL_ref(L, LUA_REGISTRYINDEX);
```

Comment:
I note that this (like the other assignments above) can leak memory if there was a previously set callback. This can happen if you call init twice... The second time it detects that it was already initialized, and skips the initializtion of the state block. Then it overwrites the pointers. Yes, it will return an error, but it is too late then.